### PR TITLE
Normative: Allow non-402 implementations to support non-iso8601 calendars

### LINKED ISSUE #2192: Should built-in calendars of non-402 implementations be limited to "iso8601"?
It seems strange to me that an ECMA-262 implementation is prohibited from supporting e.g. "gregory" or "islamicc" calendars unless it _also_ implements all requirements of ECMA-402, but that is the current state of https://tc39.es/proposal-temporal/#sec-temporal-isbuiltincalendar .

### PR DESCRIPTION
**Pull request by gibson042**
Fixes #2192

### CODE REVIEW

File: spec/calendar.html [hunk 1]
```diff
@@ -58,9 +58,9 @@ <h1>
         An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the AvailableCalendars abstract operation as specified in the ECMA-402 specification.
         If an ECMAScript implementation does not include the ECMA-402 API the following specification of the AvailableCalendars abstract operation is used.
       </p>
-      <emu-alg>
-        1. Return « *"iso8601"* ».
-      </emu-alg>
+      <p>
+        This function returns a List of String values representing calendar types supported by the implementation. The contents of the List are implementation-defined, but must contain *"iso8601"* and must not contain any String value that does not identify a calendar type in the <a href="https://cldr.unicode.org/">Unicode Common Locale Data Repository (CLDR)</a>.
```
[the remark this answers is not in the document]
Comment (gibson042) [reply spec/calendar.html]: Sure, done.

**Comment by ljharb** [comment]
LGTM!

**Review by ptomato — APPROVED** [review]
:+1:, assuming we do want to propose this normative change — I'll comment on #2192.

**Comment by ptomato** [comment]
Draft until presented to TC39 plenary.

**Review by sffc — APPROVED** [review]

**Pull request merged**